src/v/cloud_storage_clients/abs_client.cc

```diff
@@ -841,4 +873,88 @@ abs_client::do_get_account_info(ss::lowres_clock::duration timeout) {
     }
 }
 
+ss::future<> abs_client::do_delete_file(
+  const bucket_name& name,
+  object_key path,
+  ss::lowres_clock::duration timeout) {
+    vassert(
+      _adls_client && _data_lake_v2_client_config,
+      "Attempt to use ADLSv2 endpoint without having created a client");
+
+    auto header = _requestor.make_delete_file_request(
+      _data_lake_v2_client_config->uri, name, path);
+    if (!header) {
+        vlog(
+          abs_log.warn, "Failed to create request header: {}", header.error());
+        throw std::system_error(header.error());
+    }
+
+    vlog(abs_log.trace, "send https request:\n{}", header.value());
+
+    auto response_stream = co_await _adls_client->request(
+      std::move(header.value()), timeout);
+
+    co_await response_stream->prefetch_headers();
+    vassert(response_stream->is_header_done(), "Header is not received");
+
+    const auto status = response_stream->get_headers().result();
+    if (
+      status != boost::beast::http::status::accepted
+      && status != boost::beast::http::status::ok) {
+        const auto content_type = get_response_content_type(
+          response_stream->get_headers());
+        auto buf = co_await util::drain_response_stream(
+          std::move(response_stream));
+        throw parse_rest_error_response(content_type, status, std::move(buf));
+    }
+}
+
+ss::future<result<abs_client::no_response, error_outcome>>
+abs_client::delete_path(
+  bucket_name const& name,
```

Comment:
nit: curious why not `const bucket_name&` like elsewhere

Reply:
Variety is the spice of life, no? Jokes aside, no reason - will update.

Reply:
Done